### ISSUE TITLE
Use readdir instead of readdir_r

### ISSUE THREAD

**Issue by Armagetron** [googlei18n/libphonenumber#1307]
I use an Arch Linux system with gcc 6.1.1 and glibc 2.24. When building the C++ version of the library I get the following error:

```
libphonenumber/tools/cpp/src/cpp-build/generate_geocoding_data.cc: In Funktion »bool i18n::phonenumbers::ListDirectory(const string&, std::vector<i18n::phonenumbers::DirEntry>*)«:
libphonenumber/tools/cpp/src/cpp-build/generate_geocoding_data.cc:108:21: Fehler: »int readdir_r(DIR*, dirent*, dirent**)« ist veraltet [-Werror=deprecated-declarations]
     const int res = readdir_r(dir, &entry, &dir_result);
                     ^~~~~~~~~
externals/libphonenumber/tools/cpp/src/cpp-build/generate_geocoding_data.cc:19:0:
/usr/include/dirent.h:183:12: Anmerkung: hier deklariert
 extern int readdir_r (DIR *__restrict __dirp,
            ^~~~~~~~~
libphonenumber/tools/cpp/src/cpp-build/generate_geocoding_data.cc:108:55: Fehler: »int readdir_r(DIR*, dirent*, dirent**)« ist veraltet [-Werror=deprecated-declarations]
     const int res = readdir_r(dir, &entry, &dir_result);
                                                       ^
libphonenumber/tools/cpp/src/cpp-build/generate_geocoding_data.cc:19:0:
/usr/include/dirent.h:183:12: Anmerkung: hier deklariert
 extern int readdir_r (DIR *__restrict __dirp,
```

readdir_r is deprecated.

**Comment by ginggs** [comment]
This should be fixed by #1090

**Comment by penmetsaa** [comment]
@ginggs,
If you are using Boost, then #1090 will address the issue since it uses Boost instead of readdir_r.

@Armagetron,
Thanks for providing all the details we need!

Yes, readdir_r is deprecated. We may need to replace with readdir. 

However, metadata updates are the priority for us to ensure the data is fresh.
We will get to this as soon as we have some free time.
